docsite/source/rules.html.md

```diff
@@ -181,6 +181,47 @@ contract.call(email: 'jane@doe.org', login: 'jane', password: "").errors.to_h
 # => {:password=>["password is required"]}
 ```
 
+### Checking for previous errors
+
+Sometimes you may be interested in adding an error when some other error has happened.
+
+With `#schema_error?(key)` you can check whether the schema has an error for a given key:
+
+```ruby
+class PersonContract < Dry::Validation::Contract
+  schema do
+    required(:email).filled(:string)
+    required(:name).filled(:string)
+  end
+
+  rule(:name) do
+    key.failure('first introduce a valid email') if schema_error?(:email)
+  end
+end
+
+PersonContract.new.(email: nil, name: 'foo').errors.to_h
+# { email: ['must be a string'], name: ['first introduce a valid email'] }
+```
+
+In complex rules you may be interested to know whether the current rule already
+had an error. For that, you can use `#rule_error?`
```

Comment:
Please don't use hard-wrapping in docs, it can actually screw up MD formatting AND it makes writing docs really annoying

Reply:
Apologies... I fixed it :sweat_smile: